src/libs/filterInputs.ts

```diff
@@ -11,59 +17,84 @@ export enum FilterTypes {
   ExcludableCheckboxGroup = 'XCheckbox',
 }
 
-interface SwitchFilter {
+type SwitchFilter = {
   type: FilterTypes.Switch;
   /** Default value */
   value: boolean;
-}
+};
 
-interface TextFilter {
+type TextFilter = {
   type: FilterTypes.TextInput;
   /** Default value */
   value: string;
-}
+};
 
-interface CheckboxFilter {
+type CheckboxFilter = {
   type: FilterTypes.CheckboxGroup;
   options: readonly FilterOption[];
   /** Default value */
   value: string[];
-}
-interface PickerFilter {
+};
+type PickerFilter = {
   type: FilterTypes.Picker;
   options: readonly FilterOption[];
   /** Default value */
   value: string;
-}
+};
 
-interface ExcludableCheckboxFilter {
+type ExcludableCheckboxFilter = {
   type: FilterTypes.ExcludableCheckboxGroup;
   options: readonly FilterOption[];
-  value: { include?: string[]; exclude?: string[] };
-}
+  /** Default value */
+  value: {
+    /** Checkboxes marked as included */
+    include?: string[];
+    /** Checkboxes marked as included */
+    exclude?: string[];
+  };
+};
+
+/**
+ * key - filter pairs
+ */
+export type Filters = Record<string, Filter<FilterTypes>>;
 
-export type Filters = Record<
-  string,
-  { label: string } & (
-    | PickerFilter
-    | CheckboxFilter
-    | TextFilter
-    | SwitchFilter
-    | ExcludableCheckboxFilter
-  )
->;
+/**
+ * Get type of a single filter type from the {@link FilterType}
+ */
+export type Filter<Type extends FilterTypes> = {
+  label: string;
+} & (Type extends FilterTypes.CheckboxGroup
+  ? CheckboxFilter
+  : Type extends FilterTypes.Picker
+    ? PickerFilter
+    : Type extends FilterTypes.Switch
+      ? SwitchFilter
+      : Type extends FilterTypes.TextInput
+        ? TextFilter
+        : Type extends FilterTypes.ExcludableCheckboxGroup
+          ? ExcludableCheckboxFilter
+          : never);
 
```

Comment:
![image](https://github.com/user-attachments/assets/18fe2545-3871-4b33-a5f4-c6e151f3a1da)

Can we make a map type?
Like this
```ts
type FilterMapping = {
  [FilterTypes.CheckboxGroup]: CheckboxFilter
}
export type Filter<Type extends FilterTypes> = {
  label: string;
} & FilterMapping[Type]
```

Reply:
I think it should work, but I'd need to test it.

Reply:
ye, just more readable code, nested code always scares people